api/src/main/java/com/github/eyefloaters/console/api/ClientFactory.java

```diff
@@ -31,30 +30,42 @@
 import org.apache.kafka.common.config.SslConfigs;
 import org.apache.kafka.common.security.auth.SecurityProtocol;
 import org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule;
-import org.apache.kafka.common.security.plain.PlainLoginModule;
 import org.apache.kafka.common.serialization.ByteArrayDeserializer;
-import org.eclipse.microprofile.jwt.JsonWebToken;
 import org.jboss.logging.Logger;
 
 import com.github.eyefloaters.console.api.service.KafkaClusterService;
-import com.github.eyefloaters.console.legacy.model.AdminServerException;
-import com.github.eyefloaters.console.legacy.model.ErrorType;
 
 import io.fabric8.kubernetes.client.informers.SharedIndexInformer;
 import io.strimzi.api.kafka.model.Kafka;
 import io.strimzi.api.kafka.model.status.ListenerStatus;
 
+/**
+ * The ClientFactory is responsible for managing the life-cycles of Kafka clients
+ * - the {@linkplain Admin} client and the {@linkplain Consumer}. The factory
+ * will create a per-request client lazily when accessed by
+ * {@linkplain com.github.eyefloaters.console.api.service service code} which
+ * will be usable for the duration of the request and closed by the disposer
+ * methods in this class upon completion of the request.
```

Comment:
Just so I'm clear, this means that right if the front end needs to make multiple requests to render some UI thne we end up creating multiple clients (and multiple client authentication flows etc), right?

Reply:
> Just so I'm clear, this means that right if the front end needs to make multiple requests to render some UI thne we end up creating multiple clients (and multiple client authentication flows etc), right?

That's correct. It's something we inherited from the RHOSAK design and something we'll need to keep an eye on from a scalability/performance perspective at some point.